src/components/Mission.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+
+import { joinMission, leaveMission } from '../redux/missions/missionsSlice';
+
+const Button = ({ isJoined, missionId }) => {
+  const dispatch = useDispatch();
+
+  if (isJoined) {
+    return (
+      <button
+        className="btnm failure__button"
+        type="button"
+        onClick={() => dispatch(leaveMission(missionId))}
+      >
+        Leave Mission
+      </button>
+    );
+  }
+  return (
+    <button
+      className=" btnm success__button"
+      type="button"
+      onClick={() => dispatch(joinMission(missionId))}
+    >
+      Join Mission
+    </button>
+  );
+};
```

Comment:
- [ ] @team it's not good practice having 2 components in the same file, kindly move the button component in its own file then import it, this will make your code more readable ad easier to scale and debug in the future.